lib/settings_base.py

```diff
@@ -80,6 +80,9 @@ def path(*folders):
 DATABASES['default']['TEST_CHARSET'] = 'utf8'
 DATABASES['default']['TEST_COLLATION'] = 'utf8_general_ci'
 
+# Run all views in a transaction unless they are decorated not to.
+ATOMIC_REQUESTS = True
```

Comment:
Yes. my understanding is that it would:
- open a transaction against master right before entering the view
- read from the replica
- update master
- commit the transaction once the view is complete

Do note that IIRC transactions are opened against master even if the view does not use master (or even if it does not do any database operations). That's why we want to use`non_atomic_requests` decorators wherever we think we're not going to have writes.